File: iosNativeApp/iosNativeApp/Domain/UseCase/GetCurrencyRatesUseCase.swift
final class GetCurrencyRatesUseCase {
    private let repository: CurrencyRepository

    init(repository: CurrencyRepository) {
        self.repository = repository
    }

    func execute() async -> Result<[String: Double], Error> {
        return await repository.getCurrencyRates()
    }
}

File: sharedUI/src/commonMain/kotlin/io/github/currencyconverter/domain/usecase/GetCurrencyRatesUseCase.kt
package io.github.currencyconverter.domain.usecase

import io.github.currencyconverter.domain.repository.CurrencyRepository

class GetCurrencyRatesUseCase(
    private val currencyRepository: CurrencyRepository
) {

    suspend operator fun invoke(): Result<Map<String, Double>> = currencyRepository.getCurrencyRates()
}